### PR TITLE
Add tensorboard configure into ./common/fit.py and ./train_mnist.py

### PR DESCRIPTION
**Pull request by ysh329**
Some users, including me, are confused about tensorboard usage, which after making tensorboard enable, it only draws accuracy curve of training set, not containing accuracy curve of evaluation set.

Besides, the pip of tensorboard for MXNet pip is still in construction, after finishing, we'll add these parameters (`train_log`, `eval_log`) into args of parser.

### CODE REVIEW

File: example/image-classification/common/fit.py [hunk 1]
```diff
@@ -168,10 +168,16 @@ def fit(args, network, data_loader, **kwargs):
 
     # callbacks that run after each batch
     batch_end_callbacks = [mx.callback.Speedometer(args.batch_size, args.disp_batches)]
+    eval_end_callbacks = []
     if 'batch_end_callback' in kwargs:
         cbs = kwargs['batch_end_callback']
         batch_end_callbacks += cbs if isinstance(cbs, list) else [cbs]
```
Comment (zihaolucky) [inline example/image-classification/common/fit.py]: Have you verified if the logging works correctly when using `Speedometer` and `LogMetricsCallback` at the same time? Both speedometer and tensorboard logging.

Comment (zihaolucky) [inline example/image-classification/common/fit.py]: Have you verified if the logging works correctly when using Speedometer and LogMetricsCallback at the same time? Both speedometer and tensorboard logging.

**Comment by zihaolucky** [comment]
I think the TensorBoard related code is correct, but we have to make this example more consistent and easy to understand, as these examples are used by almost every user.

File: example/image-classification/train_mnist.py [hunk 2]
```diff
@@ -75,5 +75,13 @@ def get_mnist_iter(args, kv):
     net = import_module('symbols.'+args.network)
     sym = net.get_symbol(**vars(args))
 
+    # tensorboard logs
+    train_log = 'logs/mnist/train'
+    eval_log = 'logs/mnist/eval'
+    batch_end_callbacks = [mx.contrib.tensorboard.LogMetricsCallback(train_log)]
```
Comment (zihaolucky) [inline example/image-classification/train_mnist.py]: If some one hasn't install tensorboard yet, it might throw an exception. Consider making it an optional feature.

**Comment by zihaolucky** [comment]
I'm going to upload the package to another PyPI this week, might consider using `tensorboard-lite`.

The next step is to move the backend to https://github.com/tensorflow/tensorboard, the experimental code could be found here https://github.com/zihaolucky/tensorboard-lite/tree/exp/add-tf-as-local-dep.

File: example/image-classification/train_mnist.py (hunk 2 above)
Comment (ysh329) [inline example/image-classification/train_mnist.py]: Okay

Comment (zihaolucky) [inline example/image-classification/train_mnist.py]: @ysh329 You don't have to close it, just submit another commit and it would be great.

Comment (ysh329) [inline example/image-classification/train_mnist.py]: @zihaolucky I found another problem :sob: 
```Shell
root@65ea267b5b52:~/incubator-mxnet-0.10.0.post2/example/image-classification# python train_mnist.py 
INFO:root:start with arguments Namespace(add_stn=False, batch_size=64, disp_batches=100, dtype='float32', gpus=None, kv_store='device', load_epoch=None, lr=0.05, lr_factor=0.1, lr_step_epochs='10', model_prefix=None, mom=0.9, monitor=0, network='mlp', num_classes=10, num_epochs=20, num_examples=60000, num_layers=None, optimizer='sgd', test_io=0, top_k=0, wd=0.0001)
Traceback (most recent call last):
  File "train_mnist.py", line 87, in <module>
    eval_end_callback = eval_end_callbacks)
  File "/root/incubator-mxnet-0.10.0.post2/example/image-classification/common/fit.py", line 197, in fit
    monitor            = monitor)
  File "/mxnet/python/mxnet/module/base_module.py", line 465, in fit
    optimizer_params=optimizer_params)
  File "/mxnet/python/mxnet/module/module.py", line 478, in init_optimizer
    **optimizer_params)
  File "/mxnet/python/mxnet/optimizer.py", line 128, in create_optimizer
    return Optimizer.opt_registry[name.lower()](**kwargs)
  File "/mxnet/python/mxnet/optimizer.py", line 328, in __init__
    super(SGD, self).__init__(**kwargs)
TypeError: __init__() got an unexpected keyword argument 'multi_precision'
```

**Comment by zihaolucky** [comment]
@piiswrong Could you give some suggestions?

File: example/image-classification/common/fit.py [hunk 3]
```diff
@@ -168,10 +168,16 @@ def fit(args, network, data_loader, **kwargs):
...
 
+    # tensorboard logs
+    if "eval_end_callback" in kwargs:
+	cbs = kwargs['eval_end_callback']
+        eval_end_callbacks += cbs if isinstance(cbs, list) else [cbs]
```
Comment (piiswrong) [inline example/image-classification/common/fit.py]: why not assign directly here?

Comment (piiswrong) [inline example/image-classification/common/fit.py]: indentation.
Have you verified that this runs?

Comment (ysh329) [inline example/image-classification/common/fit.py]: I verified this in current master branch, but there's some problem about `common/fit.py` as below:
train_mnist.py failed: TypeError: __init__() got an unexpected keyword argument 'multi_precision' · Issue #7424 · apache/incubator-mxnet
https://github.com/apache/incubator-mxnet/issues/7424

I verified that runs in MXNet 0.10.0 Release is okay.

File: example/image-classification/common/fit.py (hunk 1 above)
Comment (ysh329) [inline example/image-classification/common/fit.py]: @zihaolucky 
I assigned `eval_end_callbacks` with same value as `batch_end_ballbacks`, but log about validation set still print for each epoch, not `args.disp_batches`.
```Python
     batch_end_callbacks = [mx.callback.Speedometer(args.batch_size, args.disp_batches)]
    eval_end_callbacks = [mx.callback.Speedometer(args.batch_size, args.disp_batches)]
```

File: example/image-classification/common/fit.py (hunk 3 above)
Comment (ysh329) [inline example/image-classification/common/fit.py]: modified as direct way

File: example/image-classification/train_mnist.py (hunk 2 above)
Comment (ysh329) [inline example/image-classification/train_mnist.py]: So stupid I am, I forgot `make` and `pip install -e .`. It's okay on current master branch.

File: example/image-classification/common/fit.py (hunk 3 above)
Comment (ysh329) [inline example/image-classification/common/fit.py]: So stupid I am, I forgot `make` and `pip install -e .`. I tested and verified this PR. It's okay on current master branch.

**Comment by piiswrong** [comment]
closing due to inactive. @zihaolucky Please reopen if this is still relevant.

Please remove the log files (too many lines).
